### ISSUE TITLE
Procedural Oriented Programming in Swoole

### ISSUE THREAD

**Issue by jobs-git** [swoole/swoole-src#2118]
How do we implement a procedural oriented programming of the following?

`
$http = new swoole_http_server("127.0.0.1", 9501);
$http->on('request', function ($request, $response) {
    $response->end("<h1>Hello Swoole. #".rand(1000, 9999)."</h1>");
});
$http->start();
`

**Comment by flddr** [comment]
What have you tried so far?

**Comment by flddr** [comment]
Btw: you can ask everything in the slack channel: https://www.swoole.co.uk/community

Maybe easier for you to have a good start with swoole

**Comment by jobs-git** [comment]
Thanks, my request to join there is not being accepted yet

**Comment by flddr** [comment]
Have you checked _spam_? It happens sometimes...

ping @doubaokun

**Comment by flddr** [comment]
While waiting you can start procedural like this

```php
$http = new swoole_http_server("127.0.0.1", 9501);
$http->on('request', function ($request, $response) {
    $unused1 = f2();
    $response->end(f1());
    $unused2 = f1();
});
$http->start();
function f1() {return f2();}
function f2() {return "<h1>Hello Swoole. #".rand(1000, 9999)."</h1>";}
```

**Comment by jobs-git** [comment]
I check but there is none in there, I registered in slack group a few days ago.

The code you shared looks interesting, but isn't `$http` still declared as an object?

How do we implement `swoole_http_server()` as procedural similar to the [mysqli in php ](http://php.net/manual/en/mysqli.prepare.php)

**Comment by flddr** [comment]
> but isn't $http still declared as an object?

Yes, like everything in swoole. But you can work as you want above this.

The underlying itself is reactor-based eventloop which calls _on_ in this example and gives _request_ and _response_ objects you can work with (or on).

Can you give an example code to better understand?

**Comment by jobs-git** [comment]
I am expecting it to be like this:

```
$http = new swoole_http_server("127.0.0.1", 9501);
swoole_http_request($http, $request = f1(), $response);
function f1(){
     return "Hello world!";
}
```

**Comment by flddr** [comment]
This is not possible. You can go as i wrote :wink:
